Switch from tslint to eslint

felix-hoc opened the pull request:
## Description
<!-- Add a short description about the changes -->

`tslint` is deprecated. Let's switch to `eslint`.

## Changes
- Add eslint dependencies and config, based on the recommended rules
- Apply quick fixes where possible
- Disable rules that cause many errors
- Remove `tslint` dependencies
- Enable `skipLibCheck` in tsconfig since there is some incompatibility between types (could not drill down exactly where it comes from)

## Checklist (for PR submitter and reviewers)
<!-- A PR with CHANGELOG entry will be released automatically -->
<!-- This is required and should be added in every case -->
- [x] `CHANGELOG` entry

felix-hoc committed:
Add eslint dependencies and config

felix-hoc committed:
Fix linting issues

felix-hoc committed:
Skip lib check to fix issues with incompatible types in dependencies

felix-hoc committed:
Enable type checks in eslint

Disable rules that cannot be fixed quickly

felix-hoc committed:
Fix linting errors

felix-hoc committed:
Disable linting checks

felix-hoc committed:
Fix linting error

replace for..in loop

felix-hoc committed:
Replace tslint with eslint in build

felix-hoc committed:
Remove `tslint` dependencies

felix-hoc committed:
Merge branch 'develop' into feature/switch-to-eslint

felix-hoc committed:
Add changelog entry

felix-hoc committed:
Delete tslint config

bitmovin-kenny commented on src/ts/components/listselector.ts:
```diff
@@ -77,12 +77,7 @@ export abstract class ListSelector<Config extends ListSelectorConfig> extends Co
-      }
-    }
-    return -1;
+    return this.items.findIndex((item) => item.key === key);
```
Just to be sure: Is this supported on old TVs?

felix-hoc replied:
I thought so, since we already use it a couple of times. But according to MDN it's only available since Chrome 45 😳 Let me run some tests.

felix-hoc replied:
I rewrote the function to not use `findIndex` :)

bitmovin-kenny commented on eslint.config.mjs:
```diff
@@ -0,0 +1,49 @@
+  ],
+  rules: {
+    'no-prototype-builtins': 'off',
+    'prefer-const': 'off', // TODO: enable rule and run autofix
```
I don't think this TODO should go into main... can you remove it now, did you run autofix?

felix-hoc replied:
There are a ton of changes that this would produce - so I'd rather do it in an isolated PR. Since https://github.com/bitmovin/bitmovin-player-ui/pull/656 has a lot of changes, I'd also like to avoid too many conflicts for them.

bitmovin-kenny replied:
OK that is very considerate of you.

bitmovin-kenny commented on eslint.config.mjs:
```diff
@@ -0,0 +1,49 @@
+// @ts-check
```
why is this here?

felix-hoc replied:
It enables type checking in a JavaScript file, which is otherwise disabled in the tsconfig. It was part of the default config, and I thought it made sense to keep it :)

bitmovin-kenny reviewed (changes requested):
Very nice 💪 
Just a few very minor nits.

felix-hoc committed:
Replace usage of `findIndex`

Not supported on older TVs, and code path may be executed before the player installs the polyfill

bitmovin-kenny reviewed (approved):
👍
